### ISSUE TITLE
unexpected error when accessing a declared ODocument's field in a live query callback

### ISSUE THREAD

**Issue by dominioon** [orientechnologies/orientdb#6836]
## OrientDB Version, operating system, or hardware.
-  [x] v2.2.11
## Operating System
- [X] Linux
### Expected behavior

`Got live update 42`
### Actual behavior

```
WARNI Error deserializing record with id #17:0 send this data for debugging: ABJMaXZlVGFibGUvAAAAEQAENDI=  [ORecordSerializerBinary]com.orientechnologies.orient.core.exception.ODatabaseException: The database instance is not set in the current thread. Be sure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db);
    at com.orientechnologies.orient.core.db.ODatabaseRecordThreadLocal.get(ODatabaseRecordThreadLocal.java:51)
    at com.orientechnologies.orient.core.record.ORecordAbstract.getDatabase(ORecordAbstract.java:241)
    at com.orientechnologies.orient.core.record.impl.ODocument.getGlobalPropertyById(ODocument.java:2197)
    at com.orientechnologies.orient.core.record.impl.ODocumentInternal.getGlobalPropertyById(ODocumentInternal.java:64)
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.getGlobalProperty(ORecordSerializerBinaryV0.java:509)
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserializePartial(ORecordSerializerBinaryV0.java:121)
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.fromStream(ORecordSerializerBinary.java:78)
    at com.orientechnologies.orient.core.record.impl.ODocument.deserializeFields(ODocument.java:1851)
    at com.orientechnologies.orient.core.record.impl.ODocument.checkForFields(ODocument.java:2559)
    at com.orientechnologies.orient.core.record.impl.ODocument.rawField(ODocument.java:773)
    at com.orientechnologies.orient.core.record.impl.ODocument.field(ODocument.java:823)
    at com.Application$1.onLiveResult(Application.java:24)
    at com.orientechnologies.orient.core.sql.query.OLocalLiveResultListener.onLiveResult(OLocalLiveResultListener.java:35)
    at com.orientechnologies.orient.client.remote.OStorageRemoteAsynchEventListener.onRequest(OStorageRemoteAsynchEventListener.java:87)
    at com.orientechnologies.orient.client.remote.ORemoteConnectionPushListener.onRequest(ORemoteConnectionPushListener.java:55)
    at com.orientechnologies.orient.client.binary.OAsynchChannelServiceThread.execute(OAsynchChannelServiceThread.java:62)
    at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```
## Steps to reproduce the problem

Create a class and set a property (named "entry" in this example) to it. Set up live query listener for the table and try to access document.field("entry") in its onLiveResult() method. Insert a record to the table. It is done in the attached program, so just run its com.Main.
[orient_6836.zip](https://github.com/orientechnologies/orientdb/files/542009/orient_6836.zip)

**Comment by dominioon** [comment]
@maggiolo00, using the same db instance that was created at line 19 is not possible (`[ORecordSerializerBinary]com.orientechnologies.orient.core.exception.ODatabaseException: Database 'remote:localhost/testDb' is closed`). Trying to acquire a new one (`try (ODatabaseDocumentTx db2 = dbPool.acquire()) {...}`) seems to lead to some "deadlock" etc, as the program hangs (!). But yes, I agree that maybe there are some workarounds, but why should I have to use them? Isn't it logical that I can get the value of any field from the document sent to me via onLiveResult() method? Also the code does not give any errors when line 57 (`clazz.createProperty("entry", OType.STRING);`) is removed from Config.java.

**Comment by maggiolo00** [comment]
@dominioon 

can you share some test code?

**Comment by maggiolo00** [comment]
@dominioon 

@tglman should provide a fix soon. So no workaround is needed.

Thanks

**Comment by tglman** [comment]
hi,
just pushed the fix.

Bye

**Comment by dominioon** [comment]
Hi @tglman and @maggiolo00,
thanks for the quick fix! Ill just in case share the code and threaddump for the deadlock-like situation I mentioned earlier:
```
...
public void onLiveResult(int i, ORecordOperation oRecordOperation) throws OException {
  // this try block was added
  try (ODatabaseDocumentTx db2 = dbPool.acquire()) {
    ODocument document = oRecordOperation.getRecord().getRecord();
    System.out.println("Got live update " + document.field("entry"));
  }
}
...
```
[threadDump.txt](https://github.com/orientechnologies/orientdb/files/543702/threadDump.txt)

If this issue is now also fixed, then good, otherwise please take a look at it, too.

**Comment by tglman** [comment]
hi @dominioon,

we fixed also this deadlock-like situatlon, all will be released in 2.2.13.

Bye

**Comment by tglman** [comment]
the fix has been released, closing
